assets/javascripts/modules/modalDialog.js

```diff
@@ -0,0 +1,351 @@
+        $tabs = $(contentShowing).find('[tabindex="1"]').addBack(),
+        inc = $tabs.index($src) + (isBack ? -1 : 1),
+        $focus;
+
+      // if focus has moved out of modal-dialog bring it back
+      if (inc < 0) {
+        $focus = $tabs.last();
+      }
+      else if (inc > $tabs.length - 1) {
+        $focus = $tabs.first();
+        $container.scrollTop(0);
+      }
+      else {
+        $focus = $tabs.eq(inc);
+      }
+
+      $focus.focus();
+    }
+  });
+
+  /**
+   * Event handler for document click to return focus to modal-dialog when content showing.
+   */
+  $doc.on('click', function(e) {
+    if (isModalDialogOpen() && !$(contentShowing).find($(e.target)).length) {
+      if ($(contentShowing).find('[tabindex="1"]:focus').length) {
+        $(contentShowing).find('[tabindex="1"]:focus').focus();
+      } else {
+        $(contentShowing).focus();
+        e.stopPropagation();
+      }
+    }
+  });
+
+  /**
+   * Event handler for hiding modal-dialog & content.
+   */
+  $body.find($contents).addBack().on('click', '[data-modaldialog-action="close"]', function(e) {
+    e.preventDefault();
+    hide();
+  });
+
+  /**
+   * Event handler for showing modal-dialog & content.
+   */
+  $body.find($contents).addBack().on('click', '[data-modaldialog-action="open"]', function(e) {
+    e.preventDefault();
+    setModalDialogDimensions();
+    show(e);
+  });
+
+  /**
+   * Event handler for custom event triggered by hidden attribute / prop state change.
+   */
+  $contents.on('hiddenChange', function(e, isHidden) {
+    if (isHidden) {
+      if (!$(contentShowing).length) {
+        hide();
+      }
+    } else {
+      show();
+    }
+  });
+
+  /**
+   * Setup elements
+   */
+  function setup() {
+    $contents.each(function() {
+      var $this = $(this);
+      $this.setHidden($this.getHidden());
+    });
+
+  }
+
+  /**
+   * Fire Google Analytics event tracking
+   */
+  function fireEventTracking(gaEvent) {
+    var gaEventFields = gaEvent.split(':');
+    if (gaEventFields.length === 3) {
+      GOVUK.performance.sendGoogleAnalyticsEvent(gaEventFields[0], gaEventFields[1], gaEventFields[2]);
+    }
+  }
+
+  /**
+   * Module initialization
+   */
+  function init() {
+    if ($modalDialog.length && $contents.length) {
+      setModalDialogDimensions();
+      setup();
+    }
+    else {
+      $modalDialog.hide();
+    }
+  }
+
+  /**
+   * Set hidden properties & attributes
+   * @param {boolean} isHidden
+   */
+  $.fn.setHidden = function(isHidden) {
+    var $el = $(this);
+
+    if (!isHidden) {
+      $el.removeProp('hidden');
+      $el.removeAttr('hidden');
+    }
+    else {
+      $el.prop('hidden', true);
+
+      // add hidden as string for IE
+      $el.attr('hidden', 'true');
+    }
+
+    $el.attr('aria-hidden', isHidden);
+    $el.trigger('hiddenChange', isHidden);
+  };
+
+  /**
+   * Get hidden property or attribute
+   * @returns {boolean|string}
+   */
+  $.fn.getHidden = function() {
```

Comment:
would these functions `setHidden` and `getHidden` (if needed, see above commet) be useful outside of this module?